<?xml version="1.0"?>
<launch>  
  <arg name="shutter_time" default="100" />
  <arg name="idle_time" default="100000"/>
  <!-- <arg name="container_size" default="64"/> -->
  <arg name="lag_compensation" default="0.001"/>
  <arg name="topic" default="laser_scan"/>
  <arg name="frame" default="scanCONTROL_2600-100scanner_laser_link"/>
  <arg name="dense" default="false"/>
  <!-- <arg name="field_left" default="0.2"/>
  <arg name="field_right" default="0.2"/>
  <arg name="field_far" default="0.2"/>
  <arg name="field_near" default="0.0"/> -->
  <arg name="serial_number" default="218120116"/>
  <arg name="path_to_device_properties" default="/home/jeeva/GitHub/SIMTech_ws/src/microEpsilon_scanControl/microEpsilon_scanControl/microepsilon_scancontrol/scanCONTROLLinuxSDK0.2.3/misc"/>
  <arg name="ns" default="microepsilon"/>
  <node ns = "$(arg ns)" name="microepsilon_scancontrol_node" pkg="microepsilon_scancontrol" type="microepsilon_scancontrol_node" cwd="node" respawn="true" output="screen">
    <param name="shutter_time" type="int" value="$(arg shutter_time)"/>
    <param name="idle_time" type="int" value="$(arg idle_time)"/>
    <!-- <param name="container_size" type="int" value="$(arg container_size)"/> -->
    <param name="lag_compensation" type="double" value="$(arg lag_compensation)"/>
    <param name="topic" type="string" value="$(arg topic)"/>
    <param name="frame" type="string" value="$(arg frame)"/>
    <param name="dense" type="bool" value="$(arg dense)"/>
    <!-- <param name="field_left" type="double" value="$(arg field_left)"/>
    <param name="field_right" type="double" value="$(arg field_right)"/>
    <param name="field_far" type="double" value="$(arg field_far)"/>
    <param name="field_near" type="double" value="$(arg field_near)"/> -->
    <param name="serial_number" type="string" value="$(arg serial_number)"/>
    <param name="path_to_device_properties" type="string" value="$(arg path_to_device_properties)"/>
  </node>


  <!-- <node name="laser_scan_transformed" pkg="microepsilon_scancontrol" type="cloud_transformation_node" respawn="true" />  -->
  
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find microepsilon_scancontrol)/description/urdf/scanCONTROL_2600-100.urdf.xacro'" />
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
    <param name="use_gui" value="true"/>
  </node>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find microepsilon_scancontrol)/launch/test_description.rviz" required="true"/>


</launch>
